index.js

```diff
@@ -28,7 +29,7 @@ module.exports = opts => {
 		try {
 			report = xo.lintText(file.contents.toString(), {
 				cwd: file.cwd,
-				filename: file.relative
+				filename: path.relative(file.cwd, file.path)
```

Comment:
> gulpjs/vinyl:index.js@master#L256

Not sure what you're trying to say by linking this, but I do realize how it works underneath.

> We are all wrong, we should use absolute path.

No, but an absolute path works too.